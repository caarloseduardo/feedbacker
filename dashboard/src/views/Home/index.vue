<template>
  <custom-header
    @create-account="handleAccountCreate"
    @login="handleLogin"
  />
  <contact />
  <div class="flex justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800">feedbacker Â© 2021</p>
  </div>
</template>

<script>
import { onMounted } from 'vue'
import { useRouter } from 'vue-router'
import CustomHeader from './CustomHeader.vue'
import Contact from './Contact.vue'
import useModal from '../../hooks/useModal'

export default {
  components: { CustomHeader, Contact },
  setup () {
    const router = useRouter()
    const modal = useModal()

    onMounted(() => {
      const token = window.localStorage.getItem('token')
      if (token) {
        router.push({ name: 'Feedbacks' })
      }
    })

    function handleLogin () {
      modal.open({
        component: 'ModalLogin'
      })
    }

    function handleAccountCreate () {
      modal.open({
        component: 'ModalAccountCreate'
      })
    }

    return {
      handleLogin,
      handleAccountCreate
    }
  }
}
</script>
