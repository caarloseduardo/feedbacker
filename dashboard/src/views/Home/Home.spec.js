import Home from '.'
import { shallowMount } from '@vue/test-utils'
import { routes } from '../../router'

import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('<Home />', () => {
  it('should render home correctly', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
