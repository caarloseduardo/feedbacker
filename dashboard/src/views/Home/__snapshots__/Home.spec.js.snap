// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home correctly 1`] = `
<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>
<div class="flex justify-center py-10 bg-brand-gray">
  <p class="font-medium text-center text-gray-800">feedbacker Â© 2021</p>
</div>
`;
